src/controllers/events.js

```diff
@@ -29,3 +29,18 @@ exports.getRelatedEvents = async (req, res, next) => {
     .populate('ngo', 'name');
   return res.status(200).json({ success: true, data: relatedEvents });
 };
+
+exports.applyToEvent = async (req, res, next) => {
+  const volunteer = await Volunteer.findById(req.user._id);
+  const event = await Event.findById(req.params.id);
+
+  if (!event) {
+    return next(new ErrorResponse('No event found', 404));
+  }
+  event.pendingApplicants.push(req.user._id);
```

Comment:
Let's check if he already applied, before pushing. I mean what if the user uses this endpoint, twice?

There is a very handy method you could use: 
```  event.pendingApplicants.addToSet(req.user._id);```

This method, will only add `req.user._id` to the `pendingApplicants` if he is not already there. It prevents duplication.